kobo-uncaged/main.go

```diff
@@ -647,63 +585,60 @@ func (ku *KoboUncaged) saveDeviceInfo() error {
 	return ioutil.WriteFile(filepath.Join(ku.bkRootDir, calibreDIfile), diJSON, 0644)
 }
 
-func (ku *KoboUncaged) saveCoverImage(contentID string, thumb []interface{}) {
+func (ku *KoboUncaged) saveCoverImage(contentID string, size image.Point, imgB64 string) {
 	defer ku.wg.Done()
```

Comment:
I would probably rework the WaitGroup code and put it closer to where it is actually needed, as this is a common source of bugs.

Reply:
Not sure of the best way to handle this. Any suggestions?

It's the first time I've used WaitGroups, so I'm open to recommendations.